### PR TITLE
Update eslint settings for plugin import

### PR DESCRIPTION
**Pull request by rak-phillip**
Signed-off-by: Phillip Rak <rak.phillip@gmail.com>

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -16,7 +16,20 @@ module.exports = {
...
+    'plugin:import/typescript',
   ],
+  settings: {
+    'import/resolver': {
+      typescript: true,
+      node:       true,
+    },
+    'import/extensions': [
+      '.js',
+      '.ts',
```
Comment (mook-as) [inline .eslintrc.js:29]: I just added a `.tsx` over in #3285 :)

Comment (mook-as) [inline .eslintrc.js:29]: I just added a `.tsx` over in #3285 :)

**Review by mook-as — COMMENTED** [review]
Presumably we'll also need to fix the new import errors.

**Comment by ericpromislow** [comment]
I realize it's in draft mode, but had a look and would add this change:
```
diff --git a/.eslintrc.js b/.eslintrc.js
index 78b6a9ee..f4f66cb0 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -168,6 +168,7 @@ module.exports = {
       },
     ],
     'import/no-useless-path-segments': ['error', { noUselessIndex: true }],
+    'import/no-named-as-default-member': 'off',
   },
 };
```

**Comment by rak-phillip** [comment]
Closing because I found a more efficient way to update our config to work with eslint import plugin

**Pull request closed without merging**